Assets/Tests/Utils/Mocks/CastleAudioForTests.cs

```diff
@@ -16,6 +17,9 @@ public override void PlayHitSound()
         public override void  PlayDestroySound()
         {
             PlayDestroySoundCalled = true;
+            DestroySoundPlayed?.Invoke();
```

Comment:
Do we really need a mock class for this? We could use the power of .Raise()/.Callback() methods of Moq's Setup function in such a case